src/Columns/ColumnTuple.cpp

```diff
@@ -101,17 +101,21 @@ MutableColumnPtr ColumnTuple::cloneResized(size_t new_size) const
 
 Field ColumnTuple::operator[](size_t n) const
 {
-    return collections::map<Tuple>(columns, [n] (const auto & column) { return (*column)[n]; });
+    Field res;
+    get(n, res);
+    return res;
 }
 
 void ColumnTuple::get(size_t n, Field & res) const
 {
     const size_t tuple_size = columns.size();
-    Tuple tuple(tuple_size);
-    for (const auto i : collections::range(0, tuple_size))
-        columns[i]->get(n, tuple[i]);
 
-    res = tuple;
+    res = Tuple();
+    Tuple & res_tuple = DB::get<Tuple &>(res);
+    res_tuple.reserve(tuple_size);
+
+    for (const auto i : collections::range(0, tuple_size))
```

Comment:
`Consider to use simple loop without collections::range.`

Reply:
I've leaved it as-is in the surrounding code, but I'm ok to change it.